frontend/src/api/open-hands.ts

```diff
@@ -111,39 +149,87 @@ class OpenHands {
    * @returns Success message or error message
    */
   static async uploadFiles(
-    file: File[],
-  ): Promise<FileUploadSuccessResponse | ErrorResponse> {
+    token: string,
+    files: File[],
+  ): Promise<FileUploadSuccessResponse> {
     const formData = new FormData();
-    file.forEach((f) => formData.append("files", f));
+    files.forEach((file) => formData.append("files", file));
 
-    return request(`/api/upload-files`, {
+    const response = await fetch("/api/upload-files", {
       method: "POST",
       body: formData,
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
     });
-  }
 
-  /**
-   * Get the blob of the workspace zip
-   * @returns Blob of the workspace zip
-   */
-  static async getWorkspaceZip(): Promise<Blob> {
-    const response = await request(`/api/zip-directory`, {}, false, true);
-    return response.blob();
+    if (!response.ok) {
+      throw new Error("Failed to upload files");
+    }
+
+    const data = (await response.json()) as
+      | FileUploadSuccessResponse
+      | ErrorResponse;
+
+    if ("error" in data) {
+      throw new Error(data.error);
+    }
+
+    return data;
   }
 
   /**
    * Send feedback to the server
    * @param data Feedback data
    * @returns The stored feedback data
    */
-  static async submitFeedback(data: Feedback): Promise<FeedbackResponse> {
-    return request(`/api/submit-feedback`, {
+  static async submitFeedback(
+    token: string,
+    feedback: Feedback,
+  ): Promise<FeedbackResponse> {
+    const response = await fetch("/api/submit-feedback", {
       method: "POST",
-      body: JSON.stringify(data),
+      body: JSON.stringify(feedback),
       headers: {
         "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    });
```

Comment:
Why not use the `request` helper which does the token plumbing for us? I don't like the idea of passing `token` around the app--it adds a lot more plumbing

Reply:
Our current request handler is quite messy. A major concern is how `request` does error handling. TanStack Query already manages error by catching thrown errors, but the `request` handler introduces its own error-handling logic. This complicates integration with TanStack Query and also makes custom error handling more difficult.

IMO retrieving the token from local storage would no longer be ideal. Our updated auth context now manages the token state (to address the improper retrieval of tokens via local storage throughout the app), and we use the `useAuth` context hook within the custom query hooks that call the request function. Basically very similar to what `request` tried to do but this better follows React principles and keeps the code more consistent and maintainable.

Final note is that `request` breaks some practice by introducing `any` types and ignores some eslint rules